### ISSUE TITLE
BUG: logging stopped after first epoch of training

### ISSUE THREAD

**Issue by niansong1996** [neptune-ai/neptune-client#861]
### Describe the bug
I used neptune to log for pytorch-lightning a few months ago and it was working okay. But today when I tried to reuse it (with everything updated), I met with this weird issue where the logging stopped after first epoch of training, *but the training is running fine*. 

So my suspicion is that since lightning is saving a model checkpoint file at the end of each epoch, which is pretty large, is it possible that by default, Neptune wants to upload that checkpoint file as well? That might be the reason that the logging is stuck and soon marked inactive.
### Reproduction
Not sure how to reproduce this and I failed to get the solution from docs or online search.
### Expected behavior
Neptune keeps logging after first epoch
### Traceback
No error is reported and not traceback is outputted, as a matter of fact, the model training is still running fine.
### Environment
**The output of `pip list`:**
```
absl-py==1.0.0
aiohttp==3.8.1
aiosignal==1.2.0
astunparse==1.6.3
async-timeout==4.0.2
attrs==21.4.0
boto3==1.21.15
botocore==1.24.15
bravado==11.0.3
bravado-core==5.17.0
cachetools==5.0.0
certifi==2021.10.8
charset-normalizer==2.0.12
click==8.0.4
comet-ml==3.28.0
configobj==5.0.6
deepspeed==0.5.10
docstring-parser==0.13
dulwich==0.20.33
everett==3.0.0
filelock==3.6.0
frozenlist==1.3.0
fsspec==2022.1.0
future==0.18.2
gitdb==4.0.9
GitPython==3.1.27
google-auth==2.6.0
google-auth-oauthlib==0.4.6
grpcio==1.44.0
hjson==3.0.2
huggingface-hub==0.4.0
idna==3.3
importlib-metadata==4.11.1
importlib-resources==5.4.0
jmespath==0.10.0
joblib==1.1.0
jsonargparse==4.2.0
jsonref==0.2
jsonschema==4.4.0
Markdown==3.3.6
monotonic==1.6
msgpack==1.0.3
multidict==6.0.2
neptune-client==0.15.1
ninja==1.10.2.3
nltk==3.7
numpy==1.22.2
nvidia-ml-py3==7.352.0
oauthlib==3.2.0
overrides==6.1.0
packaging==21.3
pandas==1.4.1
Pillow==9.0.1
protobuf==3.19.4
psutil==5.9.0
py-cpuinfo==8.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pyDeprecate==0.3.1
PyJWT==2.3.0
pyparsing==3.0.7
pyrsistent==0.18.1
python-dateutil==2.8.2
pytorch-lightning==1.5.10
pytz==2021.3
PyYAML==6.0
regex==2022.1.18
requests==2.27.1
requests-oauthlib==1.3.1
requests-toolbelt==0.9.1
rouge-score==0.0.4
rsa==4.8
s3transfer==0.5.2
sacremoses==0.0.47
scipy==1.8.0
semantic-version==2.9.0
simplejson==3.17.6
six==1.16.0
smmap==5.0.0
swagger-spec-validator==2.7.4
tensorboard==2.8.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tokenizers==0.11.5
torch==1.10.2+cu113
torchmetrics==0.7.2
tqdm==4.62.3
transformers==4.16.2
tree-sitter==0.19.0
triton==1.0.0
typing-extensions==4.1.1
typing-utils==0.1.0
urllib3==1.26.8
websocket-client==1.3.1
Werkzeug==2.0.3
wrapt==1.13.3
wurlitzer==3.0.2
yarl==1.7.2
zipp==3.7.0
```
**The operating system you're using:**
**The output of `python --version`: Python 3.8.10**

**Comment by kamil-kaczmarek** [comment]
hey @niansong1996,

Sorry for the late reply :pray: 

Thanks for reporting this, few questions:

- Did the logging ever re-started?
- Did your training eventually finished?

When you log a lot of metrics and your model file is large (by default Neptune log model checkpoints and mimic `ModelCheckpoint` behavior), you can observe delay between state of computations and state that you see in the UI. This is because Neptune send metrics and other metadata in packages.

As you did not have any error, you eventually you should have all metadata logged correctly.

**Comment by kamil-kaczmarek** [comment]
I'm closing this for now.

If the issue re-occur again, please feel free to re-open it. I'm happy to have a closer look at it :)

**Comment by wm-mask** [comment]
I am having the same issues as well, for most of my Torch Lightning detection trainings, it stops logging after one epoch and the status on the UI turns to "Inactive". However, the training keeps going (other epochs running, it's just that the metrics are not logged anymore).

**Comment by SiddhantSadangi** [comment]
Hello @wm-mask ,

Are you uploading model checkpoints after each epoch? In case the checkpoints are large, the upload to Neptune servers might take a while, depending on your connection speed, and other logging will be queued behind the upload.

If you do not want to upload checkpoints, you can disable it by setting `log_model_checkpoints=False` while initializing `NeptuneLogger()`: https://docs.neptune.ai/integrations/lightning/#__tabbed_2_1

If you want to upload the final model, you can do so after the training is complete using Neptune's `upload()` method: https://docs.neptune.ai/integrations/lightning/#logging-after-fitting-or-testing-is-finished

Please let me know if this helps.

**Comment by wm-mask** [comment]
> Are you uploading model checkpoints after each epoch? 

Hm, no. I am saving the checkpoints locally after each epoch, but I am not explicitly uploading them to Neptune.
Is there an automatic behavior that uploads the ckpt for you that I should disable? If not, then I am not doing it manually.

**Comment by SiddhantSadangi** [comment]
Yes, by default, we upload model checkpoints to Neptune. You can turn this off by by setting `log_model_checkpoints=False` while initializing `NeptuneLogger()`: https://docs.neptune.ai/integrations/lightning/#__tabbed_2_1

**Comment by wm-mask** [comment]
Thank you, it's very clear!
Also, what do you suggest as a best practice in terms of ckpt uploads? How often should I upload them to make sure I do not only save the last model?

For past trainings, does that also mean the metrics will eventually get updated on the UI once the workers on Neptune's side consumes the logs from the queue? Will that happen even if the status of the run is "Inactive"?

Thanks for your help!

**Comment by SiddhantSadangi** [comment]
> what do you suggest as a best practice in terms of ckpt uploads? How often should I upload them to make sure I do not only save the last model?

That would depend on many factors. 
Are the checkpoints just a backup in case you want to restore an interrupted training? If that is the case, then they will be saved locally even if not uploaded to Neptune, and you can still resume training. 
Also, note that large checkpoints will eat into your storage quota.
A middle ground would be to only upload checkpoints at specific intervals, like only every 10 epochs. Lightning saves all checkpoints, so you'll need to conditionally upload them based on the epoch number.

> For past trainings, does that also mean the metrics will eventually get updated on the UI once the workers on Neptune's side consumes the logs from the queue? Will that happen even if the status of the run is "Inactive"?

Ideally yes, however, the runs should not show as "Inactive" if there's a connection open between the neptune client and the server. Can you check if the status shows a failure?
You can use the `neptune sync` CLI command to sync all your local runs to the server in case of a connection issue: https://docs.neptune.ai/api/neptune_sync/
